Source/Initialization/WarpXInitData.cpp

```diff
@@ -1200,3 +1218,143 @@ void WarpX::InitializeEBGridData (int lev)
     amrex::ignore_unused(lev);
 #endif
 }
+
+void
+WarpX::ReadExternalFieldsFromFile (std::string read_from_path, MultiFab* mf,
+std::string F_name, std::string F_component)
+{
+
+    auto& warpx = WarpX::GetInstance();
+    amrex::Geometry const& geom = warpx.Geom(0);
+    const amrex::RealBox& real_box = geom.ProbDomain();
+    const auto dx = geom.CellSizeArray();
+    amrex::IntVect nodal_flag = mf->ixType().toIntVect();
+
+    // Loop over boxes.
+    for (MFIter mfi(*mf, TilingIfNotGPU()); mfi.isValid(); ++mfi)
+    {
+
+        auto series = openPMD::Series(read_from_path, openPMD::Access::READ_ONLY);
+        auto iseries = series.iterations.begin()->second;
+        auto F = iseries.meshes[F_name];
+        auto offset = F.gridGlobalOffset();
```

Comment:
Here and below: what are the `dataOrder` and the `axisLabels` of `F`?
I am asking, because `real_box` and `dx` below are `0=x`, `1=y`, `2=z`, while the data you receive here might be likely in opposite order. You need to check/assert these two attributes to be sure.